### PR TITLE
feat: more searchBox work

### PR DESCRIPTION
**Pull request by vvo**
+ do no send the full helper to widgets, bind functions
+ switch from widgets.searchbox to widgets.searchBox namespace (start with camelcase right now, avoid bad names later on)
+ allow adding classes to the generated input
+ add some doc
+ change folder requires to have a trailing slash

### CODE REVIEW

**Comment by vvo** [comment]
> or we could use bindall

Yes we could do that at some point if needed. This is more of a reference issue (having the same "search" binded function accross widgets) than a performance issue since the binding is only done at instantiation.

File: components/SearchBox/index.js [hunk 1]
```diff
@@ -1,22 +1,20 @@
...
 var cx = require('classnames');
 
-var AlgoliasearchHelper = require('algoliasearch-helper/src/algoliasearch.helper');
-var SearchResults = require('algoliasearch-helper/src/SearchResults');
-
 var SearchBox = React.createClass({
   propTypes: {
-    helper: React.PropTypes.instanceOf(AlgoliasearchHelper),
-    results: React.PropTypes.instanceOf(SearchResults),
-    onFocus: React.PropTypes.func,
```
Comment (vvo) [inline components/SearchBox/index.js]: Sorry I removed onFocus without any specific commit. This was done to clean the widget API before we decide we do need an onFocus prop and expose it in the widget.

**Comment by redox** [comment]
Sorry I'm a bit late on that one. I'm still wondering if `searchBox` is the correct name, what about `searchInput`?

**Comment by vvo** [comment]
Don't have any preference.

Maybe searchInput is more tied to the underlying HTML element we use while searchBox is not.

**Comment by redox** [comment]
Actually, "search box" seems very OK from a Google Fight POV. Nevermind!

**Comment by vvo** [comment]
Google driven API design

**Pull request merged**